dd-sdk-android/src/main/kotlin/com/datadog/android/rum/internal/domain/RumEventSerializer.kt

```diff
@@ -0,0 +1,139 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-2020 Datadog, Inc.
+ */
+
+package com.datadog.android.rum.internal.domain
+
+import com.datadog.android.core.internal.domain.Serializer
+import com.datadog.android.rum.internal.RumFeature
+import com.google.gson.JsonArray
+import com.google.gson.JsonNull
+import com.google.gson.JsonObject
+import com.google.gson.JsonPrimitive
+import java.io.PrintWriter
+import java.io.StringWriter
+import java.text.SimpleDateFormat
+import java.util.Date
+import java.util.Locale
+import java.util.TimeZone
+
+internal class RumEventSerializer : Serializer<RumEvent> {
+
+    private val simpleDateFormat = SimpleDateFormat(ISO_8601, Locale.US).apply {
+        timeZone = TimeZone.getTimeZone("UTC")
+    }
+
+    // region Serializer
+
+    override fun serialize(model: RumEvent): String {
+        val root = JsonObject()
+
+        // Event Context
+        val context = model.context
+        root.addProperty(TAG_APPLICATION_ID, context.applicationId.toString())
+        root.addProperty(TAG_SESSION_ID, context.sessionId.toString())
+        root.addProperty(TAG_VIEW_ID, context.viewId.toString())
+        root.addProperty(TAG_SERVICE, RumFeature.serviceName)
+        if (RumFeature.envName.isNotBlank()) {
+            root.addProperty(TAG_ENV, RumFeature.envName)
+        }
+
+        // Timestamp
+        val formattedDate = simpleDateFormat.format(Date(model.timestamp))
+        root.addProperty(TAG_DATE, formattedDate)
+
+        // Data
+        val eventData = model.eventData
+        root.addProperty(TAG_EVENT_CATEGORY, eventData.category)
+        when (eventData) {
+            is RumEventData.Resource -> {
+                root.addProperty(TAG_DURATION, eventData.durationNanoSeconds)
+                root.addProperty(TAG_RESOURCE_KIND, eventData.kind.value)
+                root.addProperty(TAG_HTTP_URL, eventData.url)
+            }
+            is RumEventData.UserAction -> {
+                root.addProperty(TAG_EVENT_NAME, eventData.name)
+            }
+            is RumEventData.View -> {
+                root.addProperty(TAG_RUM_DOC_VERSION, eventData.version)
+                root.addProperty(TAG_VIEW_URL, eventData.url.replace('.', '/'))
```

Comment:
here we will have problems with obfuscated classes. I already raised this issue during our brainstorming meeting this morning....there's one thing we can do about it on the dashboard by allowing the user to provide us the "mapping.txt" file.